Adds procedures for configuring private storage endpoints with Azure (#69024)
openshift/openshift-docs: modules/registry-configuring-private-storage-endpoint-azure.adoc

// Module included in the following assemblies:
//
// * post_installation_configuration/configuring-private-cluster.adoc

:_mod-docs-content-type: CONCEPT
[id="registry-configuring-private-storage-endpoint-azure_{context}"]
= Configuring a private storage endpoint on Azure

> The Image Registry Operator can be leveraged to use private endpoints on Azure, which enables seamless configuration of private storage accounts when {product-title} is deployed on private Azure clusters. This allows users to deploy the image registry without exposing public-facing storage endpoints.

Reviewer: Suggest to make it active

```suggestion
You can leverage the Image Registry Operator to use private endpoints on Azure, which enables seamless configuration of private storage accounts when {product-title} is deployed on private Azure clusters. This allows you to deploy the image registry without exposing public-facing storage endpoints.
```

Author: fixed x2

> Configuring the Image Registry Operator to use private storage endpoints on Azure can be done in one of two ways:

Reviewer: ```suggestion
You can configure the Image Registry Operator to use private storage endpoints on Azure in one of two ways:
```

* By configuring the Image Registry Operator to discover the VNet and subnet names

* With user-provided Azure Virtual Network (VNet) and subnet names

[id="limitations-configuring-private-storage-endpoint-azure"]
== Limitations for configuring a private storage endpoint on Azure 

The following limitations apply when configuring a private storage endpoint on Azure:

> * When configuring the Image Registry Operator to use a private storage endpoint, public network access to the storage account is disabled. Consequently, pulling images from the registry outside of {product-title} only works by setting `disableRedirect: true` in the registry operator configuration. With redirect enabled, the registry redirects the client to pull images directly from the storage account, which will no longer work due to disabled public network access. For more information, see "Disabling redirect when using a private storage endpoint on Azure".

Reviewer: ```suggestion
* When configuring the Image Registry Operator to use a private storage endpoint, public network access to the storage account is disabled. Consequently, pulling images from the registry outside of {product-title} only works by setting `disableRedirect: true` in the registry Operator configuration. With redirect enabled, the registry redirects the client to pull images directly from the storage account, which will no longer work due to disabled public network access. For more information, see "Disabling redirect when using a private storage endpoint on Azure".
```

// Module included in the following assemblies:
//
// * post_installation_configuration/configuring-private-cluster.adoc

:_mod-docs-content-type: PROCEDURE
[id="registry-configuring-private-storage-endpoint-azure_{context}"]
= Configuring a private storage endpoint on Azure

The Image Registry Operator can be leveraged to use private endpoints on Azure, which enables seamless configuration of private storage accounts when {product-title} is deployed on private Azure clusters. This allows users to deploy the image registry without exposing public-facing endpoints.

> //q Is this paragraph correct?

Author: Fixed!

> //q Is this paragraph correct?

Reviewer 2: Did you forget to push? It still says `without exposing public-facing endpoints.` instead of `without exposing public-facing storage endpoints.`

> Configuring the Image Registry Operator to use private endpoints on Azure can be done in one of two ways:

Author: fixed

> Configuring the Image Registry Operator to use private endpoints on Azure can be done in one of two ways:

Reviewer 2: same here, still says `private endpoints` instead of `private storage endpoints`

* By configuring the Image Registry Operator to discover the VNet and subnet names

* With user-provided Azure Virtual Network (VNet) and subnet names

[id="limitations-configuring-private-storage-endpoint-azure"]
== Limitations for configuring a private storage endpoint on Azure 

The following limitations apply when configuring a private storage endpoint on Azure:

> * When configuring the Image Registry Operator to make the registry private, public access to the network in the storage account is disabled. Consequently, when the image registry is set up with private endpoints, pulling images from the registry outside of {product-title} only works by setting `disableRedirect: true` in your `config.yaml` file. This is necessary due to the disabled public access to the network in the storage account. For more information, see "Disabling redirect when using a private storage endpoint on Azure".

Author: fixed

> * When configuring the Image Registry Operator to make the registry private, public access to the network in the storage account is disabled. Consequently, when the image registry is set up with private endpoints, pulling images from the registry outside of {product-title} only works by setting `disableRedirect: true` in your `config.yaml` file. This is necessary due to the disabled public access to the network in the storage account. For more information, see "Disabling redirect when using a private storage endpoint on Azure".

Reviewer 2: same here.

// Module included in the following assemblies:
//
// * post_installation_configuration/configuring-private-cluster.adoc

:_mod-docs-content-type: PROCEDURE
[id="registry-configuring-private-storage-endpoint-azure_{context}"]
= Configuring a private storage endpoint on Azure

> The Image Registry Operator can be leveraged to use private endpoints on Azure, which enables seamless integration with private storage accounts when {product-title} is deployed on private Azure clusters. This allows users to deploy the image registry without exposing public-facing endpoints. 

Author: I rephrased. Thanks!

//q Is this paragraph correct?

Configuring the Image Registry Operator to use private endpoints on Azure can be done in one of two ways:

* With user-provided Azure Virtual Network (VNet) and subnet names
* By configuring the Image Registry Operator to discover the VNet and subnet names

[id="limitations-configuring-private-storage-endpoint-azure"]
== Limitations for configuring a private storage endpoint on Azure 

The following limitations apply when configuring a private storage endpoint on Azure:

* When configuring the Image Registry Operator to make the registry private, public access to the network in the storage account is also disabled.

> * When setting the Image Registry Operator back to public, the private endpoints are deleted. 

Reviewer 2: Ah yes that bullet point was outdated, sorry about that - I updated the story now to reflect this.
Your suggested change looks good.

> * When the Image Registry Operator is configured with private endpoints, pulling images from the registry outside of {product-title} _only works_ by setting `disableRedirect: true` in your `config.yaml` file. This assumes that a route is configured.

Author: Should we combine them? maybe: 

```suggestion
When configuring the Image Registry Operator to make the registry private, public access to the network in the storage account is disabled. Consequently, when the Image Registry Operator is set up with private endpoints, pulling images from the registry outside of {product-title} only works by setting `disableRedirect: true `in your `config.yaml file`. This is necessary due to the disabled public access to the network in the storage account. 
```

Or we can leave the separate

> * When the Image Registry Operator is configured with private endpoints, pulling images from the registry outside of {product-title} _only works_ by setting `disableRedirect: true` in your `config.yaml` file. This assumes that a route is configured.

Reviewer 2: Combining them makes sense to me!
